File: VB/MapErrorEvent/Form1.vb
Imports DevExpress.XtraMap
Imports System
Imports System.Windows.Forms

Namespace MapErrorEvent
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim imageLayer As New ImageLayer()
            mapControl1.Layers.Add(imageLayer)
            Dim provider As New WmsDataProvider()
            imageLayer.DataProvider = provider
            provider.ServerUri = "http://YOUR_SERVER_URI"
            provider.ActiveLayerName = "ACTIVE_LAYER_NAME"
            AddHandler imageLayer.Error, AddressOf OnError
        End Sub
        Private Sub OnError(ByVal sender As Object, ByVal e As MapErrorEventArgs)
            MessageBox.Show(e.Exception.Message)
        End Sub
    End Class
End Namespace

File: CS/MapErrorEvent/Form1.cs
using DevExpress.XtraMap;
using System;
using System.Windows.Forms;

namespace MapErrorEvent {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            ImageLayer imageLayer = new ImageLayer();
            mapControl1.Layers.Add(imageLayer);
            WmsDataProvider provider = new WmsDataProvider();
            imageLayer.DataProvider = provider;
            provider.ServerUri = "http://YOUR_SERVER_URI";
            provider.ActiveLayerName = "ACTIVE_LAYER_NAME";
            imageLayer.Error += OnError;
        }
        private void OnError(object sender, MapErrorEventArgs e) {
            MessageBox.Show(e.Exception.Message);
        }
    }
}
